src/tqec/compile/tree/node.py

```diff
@@ -41,11 +41,23 @@ def __init__(
                 no annotations are provided.
         """
         self._layer = layer
-        self._children = LayerNode._get_children(layer)
+        # If the node is a leaf node, a time order can be assigned to it.
+        # The order is used to determine the order in which the layer happens
+        # in the final circuit.
+        self._time_order: int | None = None
+        self._children: list[LayerNode] = self._create_children(layer)
         self._annotations = dict(annotations) if annotations is not None else {}
 
+    @property
+    def order(self) -> int | None:
+        return self._time_order
+
+    @order.setter
+    def order(self, value: int | None) -> None:
+        self._time_order = value
+
     @staticmethod
-    def _get_children(layer: LayoutLayer | BaseComposedLayer) -> list[LayerNode]:
+    def _create_children(layer: BaseLayer | BaseComposedLayer) -> list[LayerNode]:
```

Comment:
Is there a reason why the type annotation changed here? This method is only called in `Layer.__init__`, which has a `LayoutLayer | BaseComposedLayer` annotation. We only expect `LayoutLayer` as leaves in the `LayerTree`.